app/actions/contactsActions.js

```diff
@@ -5,13 +5,79 @@ import { has, isEmpty, keys, values, indexOf, zipObject, omit } from 'lodash-es'
 
 import { getStorage, setStorage } from '../core/storage'
 
+export const ID = 'contacts'
+
+export type TChains = { contactKey: string, chain: string }[]
+
+const STORAGE_KEY_CHAIN = 'chainBook'
+const STORAGE_KEY = 'addressBook'
+
+export const getChains = async (): Promise<TChains> => {
+  const chains = await getStorage(STORAGE_KEY_CHAIN)
+  return Array.isArray(chains) ? chains : []
+}
+
+const setChainContact = async (contactKey: string, chain: string) => {
+  const chains = await getChains()
+  const chainContacts: TChains = Array.isArray(chains) ? chains : []
+  if (contactKey && chain) {
+    chainContacts.push({ contactKey, chain })
+    await setStorage(STORAGE_KEY_CHAIN, chainContacts)
+  }
+}
+
+const updateChainContact = async (
+  contactKey: string,
+  chain: string,
+  oldContactKey: string,
+) => {
```

Comment:
Where is the validation of n3 addresses here?

Reply:
it's in chain parameter